server/services/v1/user_metadata_provider.go

```diff
@@ -0,0 +1,120 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/rs/zerolog/log"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/request"
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+type UserMetadataProvider interface {
```

Comment:
We need some integration tests for this provider.

Reply:
we need integration testing that covers all area behind auth/namespace yes. I will take care of integration test in a couple of week.